source/auth/tests/legacy/connection-string.yml

```diff
@@ -411,46 +382,23 @@ tests:
     mechanism: MONGODB-OIDC
     mechanism_properties:
       PROVIDER_NAME: aws
-- description: should recognise the mechanism when auth source is explicitly specified
-    and with aws device (MONGODB-OIDC)
-  uri: mongodb://localhost/?authMechanism=MONGODB-OIDC&authSource=$external&authMechanismProperties=PROVIDER_NAME:aws
-  valid: true
-  credential:
-    username:
-    password:
-    source: "$external"
-    mechanism: MONGODB-OIDC
-    mechanism_properties:
-      PROVIDER_NAME: aws
-- description: should throw an exception if username and password are specified (MONGODB-OIDC)
-  uri: mongodb://user:pass@localhost/?authMechanism=MONGODB-OIDC
-  callback:
-  - oidcRequest
+- description: should throw an exception if username is specified for aws (MONGODB-OIDC)
+  uri: mongodb://principalName@localhost/?authMechanism=MONGODB-OIDC&PROVIDER_NAME:aws
   valid: false
   credential:
-- description: should throw an exception if username and deviceName are specified
-    (MONGODB-OIDC)
-  uri: mongodb://principalName@localhost/?authMechanism=MONGODB-OIDC&PROVIDER_NAME:gcp
+- description: should throw an exception if specified provider is not supported (MONGODB-OIDC)
+  uri: mongodb://localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=PROVIDER_NAME:invalid
   valid: false
   credential:
-- description: should throw an exception if specified deviceName is not supported
-    (MONGODB-OIDC)
-  uri: mongodb://localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=PROVIDER_NAME:unexisted
+- description: should throw an exception custom callback is chosen but no callback is provided (MONGODB-OIDC)
+  uri: mongodb://localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=PROVIDER_NAME:custom
```

Comment:
I suppose "custom" is not allowed value for PROVIDER_NAME. As far as I remember there is only "aws" allowed so far.

Reply:
You're right, that test is vestigial from an earlier version of the spec. I've removed this test case because it's no longer relevant.